struct_test.go

```diff
@@ -148,6 +151,7 @@ func Test_MapToStruct(t *testing.T) {
 			So(ts.Name, ShouldEqual, "Unknwon")
 			So(ts.Age, ShouldEqual, 21)
 			So(ts.Male, ShouldBeTrue)
+			So(ts.Optional, shouldEqualPtr, true)
```

Comment:
Would you mind explain why `ShouldEqual` won't work here?

Reply:
ShouldEqual will compare the address value. I've changed it for `So(*ts.Optional, ShouldBeTrue)`